### PR TITLE
F/simple structure

### LINKED ISSUE #4: Infinite loop
There is an error when a player is hit and the number of moves can not be done as the other player is occupying the space. Implement pass functionality

### LINKED ISSUE #8: Change board structure to abstract direction
Direction should be handled by the user interface rather than the internal logic. This will make it easier to handle checks in the move function

### PR DESCRIPTION
**Pull request by jscruz**
closes https://github.com/jscruz/backgammon-go/issues/8 
resolves https://github.com/jscruz/backgammon-go/issues/4

### CODE REVIEW

File: model/board.go [hunk 1]
```diff
@@ -11,68 +11,90 @@ type Player int
...
+
+	HomePosition = 0
 )
 
 // Board backgammon board
 type Board struct {
-	Turn  Player
-	Board [][]int
+	Turn  int
+	Board [Spaces][Spaces]int
```
Comment (cuotos) [inline model/board.go]: Your board is an array, 26 items large, with and array of 26 in each of them... is that correct??

or did you intend to have an array of 2, each containing 26 items.

**Review by cuotos — APPROVED** [review]

File: model/board.go [hunk 2]
```diff
@@ -11,68 +11,90 @@ type Player int
...
+			possibleMoves = append(possibleMoves, i)
+		}
+		if i-moves > HomePosition && currentPlayerBoard[i] > 0 && adversaryBoard[i-moves] < 2 {
+			possibleMoves = append(possibleMoves, i)
+		}
+	}
+	return possibleMoves
 }
 
 func (b *Board) RollDie() int {
```
Comment (cuotos) [inline model/board.go]: RollDie() doesnt need to be a receiver function as it just returns a random number, no need to be a "board" method

**Review by cuotos — APPROVED** [review]

File: model/board.go (hunk 1 above)
Comment (jscruz) [reply model/board.go]: Fixed

File: model/board.go (hunk 2 above)
Comment (jscruz) [reply model/board.go]: Fixed

**Pull request merged**